decidim-forms/app/views/decidim/forms/questionnaires/answers/_matrix_multiple.html.erb

```diff
@@ -1,44 +1,47 @@
-                                    translated_attribute(answer_option.body),
-                                    choice.present?,
-                                    id: "#{choice_id}_body", disabled: disabled %>
-              <%= hidden_field_tag "questionnaire[responses][#{answer_idx}][choices][][answer_option_id]",
-                                    answer_option.id,
-                                    id: "#{choice_id}_answer_option",
-                                    disabled: true %>
-              <%= hidden_field_tag "questionnaire[responses][#{answer_idx}][choices][][matrix_row_id]",
-                                    row.id,
-                                    id: "#{choice_id}_matrix_row",
-                                    disabled: true %>
-              <% if answer_option.free_text %>
-                <%= text_field_tag "questionnaire[responses][#{answer_idx}][choices][][custom_body]",
-                choice.try(:custom_body),
-                id: "#{choice_id}_custom_body",
-                disabled: true,
-                maxlength: maxlength %>
-              <% end %>
-            </div>
-          </td>
+          <th><%= translated_attribute(answer_option.body) %></th>
         <% end %>
       </tr>
-    <% end %>
-  </tbody>
-</table>
+    </thead>
+    <tbody>
+      <% answer.question.matrix_rows.by_position.each_with_index do |row, row_idx| %>
+        <tr class="js-check-box-collection">
+          <td><%= translated_attribute row.body %></td>
+          <% answer.question.answer_options.each_with_index do |answer_option, idx| %>
+            <% choice = answer.selected_choices.find { |choice| choice.answer_option_id == answer_option.id && choice.matrix_row_id == row.id } %>
+            <% choice_id = "#{field_id}_matrix_row_#{row_idx}_choice_#{idx}" %>
+            <td>
+              <div class="js-collection-input">
+                <%= check_box_tag "questionnaire[responses][#{answer_idx}][choices][][body]",
+                                      translated_attribute(answer_option.body),
+                                      choice.present?,
+                                      id: "#{choice_id}_body", disabled: disabled %>
+                <%= hidden_field_tag "questionnaire[responses][#{answer_idx}][choices][][answer_option_id]",
+                                      answer_option.id,
+                                      id: "#{choice_id}_answer_option",
+                                      disabled: true %>
+                <%= hidden_field_tag "questionnaire[responses][#{answer_idx}][choices][][matrix_row_id]",
+                                      row.id,
+                                      id: "#{choice_id}_matrix_row",
+                                      disabled: true %>
+                <% if answer_option.free_text %>
+                  <%= text_field_tag "questionnaire[responses][#{answer_idx}][choices][][custom_body]",
```

Comment:
The free text color contrast still does not seem quite right, although it is slightly better than last time:
![Color contrast](https://user-images.githubusercontent.com/864340/185061531-58c353f8-62fc-48ed-8dd0-50f1326004a2.png)

When the input is enabled, the border color against the white background is `#d0d3d8` which does not meet the minimum color contrast requirement of 3:1
![Color contrast border](https://user-images.githubusercontent.com/864340/185061783-9548b7ff-5509-4349-95b5-77e1ab50dde7.png)
https://webaim.org/resources/contrastchecker/?fcolor=D0D3D8&bcolor=FFFFFF

More information:
https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html

![Input border color contrast requirements](https://user-images.githubusercontent.com/864340/185061979-74e4f5ee-3df1-4c90-9768-70a17c1ebc75.png)

The same requirement of 3:1 also applies between the input background color and the border color which is right now slightly worse than against the background:
![Color contrast background](https://user-images.githubusercontent.com/864340/185062274-e0ea29d4-d8d1-46d0-b333-b19538d20a6e.png)
https://webaim.org/resources/contrastchecker/?fcolor=D0D3D8&bcolor=FAFBFC

I also think the enabled input fields have to be clearly distinguishable from the disabled state but meeting the WCAG criteria here should help improving this.

Reply:
Also on mobile, I would add a min-width to the matrix questions' free text fields:
![Mobile matrix free text](https://user-images.githubusercontent.com/864340/185064055-fb10798b-4c6d-472f-9df0-42b77423dd3b.png)

Right now it does not necessarily identify as a text element where the user can write something. Plus it may be hard to fill those fields as you can only see 1-2 characters at a time.

Reply:
> [...] meet the WCAG criteria

Since our designer is off a couple of weeks, what about postpone this matter 'til then? I've already written it down [in the pending issue](https://github.com/decidim/decidim/issues/9675). The solution is actually very simple, just editting the tailwind-config.js. In such way, we can move forward on this.

Reply:
Yes sure it is fine for me.

Do you want to look into the min-width before merging or add that to the pending issues list too?

The specs needs to be fixed before merging, though.

Reply:
Yeah, we're on it. Just pointing the WCAG issue. I presume we can make it along this morning